connectivity-exporter/packet/packet.go

```diff
@@ -232,18 +232,19 @@ func (s *NetworkDataSource) TrackConnections(ctx context.Context, wg *sync.WaitG
 }
 
 func (s *State) deleteExpiredSNIs() {
-	metrics.SNIMutex.Lock()
-	defer metrics.SNIMutex.Unlock()
-	for name, sni := range s.snis {
-		if sni.Expired {
+	now := time.Now()
```

Comment:
idea: if the time.Now() function were passed in as an argument and hence we could use a fake time provider in a unit test, we could "advance" the time without waiting for it to pass